app/components/export-tool/component.js

```diff
@@ -1,138 +1,184 @@
 import Ember from 'ember';
+import { writeCSV, squash, uniqueFields } from 'exp-models/utils/csv-writer';
 
 /**
  * @module experimenter
  * @submodule components
-  */
+ */
 
-// Make sure that ember-data objects are serialized to a JS object
+/**
+ * Make sure that ember-data objects are serialized to a JS object
+ *
+ * @param obj
+ * @returns {*}
+ */
 function serializeItem(obj) {
     if (obj.serialize) {
-        var serialized = obj.serialize();
-        obj = serialized.data.attributes;
+        obj = obj.serialize().data.attributes;
     }
     return obj;
 }
 
-// Flatten a nested object into a single level, with dotted paths for keys
-function squash(obj, prefix) {
-    var ret = {};
-    Object.keys(obj).forEach((key) => {
-        var value = Ember.get(obj, key);
-        if (value && value.toJSON) {
-            value = value.toJSON();
-        }
-
-        if (Ember.$.isPlainObject(value)) {
-            ret = Ember.$.extend({}, ret, squash(value, prefix ? `${prefix}.${key}` : key));
-        } else {
-            ret[prefix ? `${prefix}.${key}` : key] = value;
-        }
-    });
-    return ret;
-}
-
 /**
  * Export tool component: serializes records into one of a number of possible output formats
  * @class export-tool
  */
 export default Ember.Component.extend({
     /**
-     * @property data The data to be serialized
+     * @property {null|Array<Object>} data The data to be serialized
```

Comment:
Tangential note, not a blocker:

Yuidoc syntax for an array of items would be, eg `{Object[]}`